lib/rules/at-rule-descriptor-no-unknown/index.cjs

```diff
@@ -0,0 +1,68 @@
+// NOTICE: This file is generated by Rollup. To modify it,
+// please instead edit the ESM counterpart and rebuild with Rollup (npm run build).
+'use strict';
+
+const cssTree = require('css-tree');
+const atKeywords = require('../../reference/atKeywords.cjs');
+const report = require('../../utils/report.cjs');
+const ruleMessages = require('../../utils/ruleMessages.cjs');
+const validateOptions = require('../../utils/validateOptions.cjs');
+
+const UNSUPPORTED_NESTING_REGEX = new RegExp(
+	`^((?!${[...atKeywords.nestingSupportedAtKeywords.values()].join('|')}).)*$`,
+	'i',
+);
+
+const ruleName = 'at-rule-descriptor-no-unknown';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (atRule, descriptor) =>
+		`Unexpected unknown descriptor "${descriptor}" for at-rule "${atRule}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/at-rule-descriptor-no-unknown',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules(UNSUPPORTED_NESTING_REGEX, (atRule) => {
+			atRule.walkDecls((decl) => {
+				const { prop, value } = decl;
```

Comment:
```suggestion
		root.walkAtRules(UNSUPPORTED_NESTING_REGEX, (atRule) => {
		    if (!isStandardSyntaxAtRule(atRule)) return;
			atRule.walkDecls((decl) => {
			    if (!isStandardSyntaxDeclaration(decl)) return;
				const { prop, value } = decl;
```

Let's return early for non-standard at-rules and decls.

See https://github.com/stylelint/stylelint/pull/8231

Reply:
Applied your suggestion!
[Return early for non-standard at-rules and decls](https://github.com/stylelint/stylelint/pull/8197/commits/f3bc4afae0d0c9d5bd9b81a38ebe037f583e4564)